jinahub/indexers/query/vector/AnnoyIndexer/__init__.py

```diff
@@ -17,43 +18,59 @@ class AnnoyIndexer(BaseNumpyIndexer):
         Annoy package dependency is only required at the query time.
     """
 
-    def __init__(self, metric: str = 'euclidean', n_trees: int = 10, search_k: int = -1, *args, **kwargs):
+    def __init__(
+            self,
+            top_k: int = 10,
+            num_dim: int = 768,
+            metric: str = 'euclidean',
+            num_trees: int = 10,
+            dump_path: str = None,
+            traverse_path: list = ['r'],
+            **kwargs,
+    ):
         """
         Initialize an AnnoyIndexer
 
+        :param top_k: get tok k vectors
+        :param num_dim: dimension of the vector
         :param metric: Metric can be "angular", "euclidean", "manhattan", "hamming", or "dot"
-        :param n_trees: builds a forest of n_trees trees. More trees gives higher precision when querying.
-        :param search_k: At query time annoy will inspect up to search_k nodes which defaults to
-            n_trees * k if not provided (set to -1)
+        :param num_trees: builds a forest of n_trees trees. More trees gives higher precision when querying.
+        :param traverse_path: traverse path on docs, e.g. ['r'], ['c']
         :param args:
         :param kwargs:
         """
-        super().__init__(*args, compress_level=0, **kwargs)
+        super().__init__(**kwargs)
+        self.top_k = top_k
         self.metric = metric
-        self.n_trees = n_trees
-        self.search_k = search_k
-
-    def build_advanced_index(self, vecs: 'np.ndarray'):
-        """Build an advanced index structure from a numpy array.
+        self.num_dim = num_dim
+        self.num_trees = num_trees
+        self.indexer = AnnoyIndex(self.num_dim, self.metric)
+        self.traverse_path = traverse_path
+        self.logger = JinaLogger(self.metas.name)
+        if dump_path is not None:
```

Comment:
Unfortunately this won't be enough. The dump path is passed to `kwargs.runtime_args` when it's coming from a rolling update. Check https://github.com/jina-ai/indexers/blob/69279e78f2e437e50a23270e915abc3d7c4a7499/jinahub/indexers/query/compound/NumpyPostgresQueryIndexer/__init__.py#L22